python/foglamp/services/south/server.py

```diff
@@ -158,13 +168,13 @@ class Server(FoglampMicroservice):
                                                                   key=data['key'],
                                                                   readings=data['readings']))
                 # pollInterval is expressed in milliseconds
-                sleep_seconds = int(config['pollInterval']['value']) / 1000.0
+                sleep_seconds = int(self._plugin_handle['pollInterval']['value']) / 1000.0
                 await asyncio.sleep(sleep_seconds)
                 # If successful, then set retry count back to 1, meaning that only in case of 3 successive failures, exit.
                 try_count = 1
             except KeyError as ex:
                 _LOGGER.exception('Keyerror plugin {} : {}'.format(self._name, str(ex)))
-            except (Exception, RuntimeError) as ex:
+            except (Exception, RuntimeError, exceptions.DataRetrievalError) as ex:
```

Comment:
plugin specific? ` exceptions.DataRetrievalError` if Yes, we should capture in pythonic way to keep things independent